api/data_refinery_api/views.py

```diff
@@ -905,205 +733,138 @@ def get(self, request, format=None):
-                                        Q(specimen_part__icontains=filter_by) |
-                                        Q(genotype__icontains=filter_by) |
-                                        Q(disease__icontains=filter_by) |
-                                        Q(disease_stage__icontains=filter_by) |
-                                        Q(cell_line__icontains=filter_by) |
-                                        Q(treatment__icontains=filter_by) |
-                                        Q(race__icontains=filter_by) |
-                                        Q(subject__icontains=filter_by) |
-                                        Q(compound__icontains=filter_by) |
-                                        Q(time__icontains=filter_by) |
-                                        Q(sampleannotation__data__icontains=filter_by)
-                                    )
-
-        page = self.paginate_queryset(samples)
-        if page is not None:
-            serializer = DetailedSampleSerializer(page, many=True)
-            return self.get_paginated_response(serializer.data)
-        else:
-            serializer = DetailedSampleSerializer(samples, many=True)
-            return Response(serializer.data)
-
-class SampleDetail(APIView):
-    """
-    Retrieve a Sample instance.
-    """
-    def get_object(self, pk):
-        try:
-            return Sample.public_objects.get(pk=pk)
-        except Sample.DoesNotExist:
-            raise Http404
+        return filter_dict
 
-    def get(self, request, pk, format=None):
-        sample = self.get_object(pk)
-        serializer = DetailedSampleSerializer(sample)
-        return Response(serializer.data)
+class SampleDetail(generics.RetrieveAPIView):
+    """ Retrieve the details for a Sample given it's accession code """
+    lookup_field = "accession_code"
+    queryset = Sample.public_objects.all()
+    serializer_class = DetailedSampleSerializer
 
 ##
 # Processor
 ##
 
-class ProcessorList(APIView):
+class ProcessorList(generics.ListAPIView):
     """List all processors."""
-    def get(self, request, format=None):
-        processors = Processor.objects.all()
-        serializer = ProcessorSerializer(processors, many=True)
-        return Response(serializer.data)
+    queryset = Processor.objects.all()
+    serializer_class = ProcessorSerializer
 
 
 ##
 # Results
 ##
 
-class ResultsList(PaginatedAPIView):
+class ComputationalResultsList(generics.ListAPIView):
     """
-    List all ComputationalResults.
+    computational_results_list
 
-    Append the pk to the end of this URL to see a detail view.
+    This lists all `ComputationalResult`. Each one contains meta-information about the output of a computer process. (Ex Salmon).
 
+    This can return valid S3 urls if a valid [token](#tag/token) is sent in the header `HTTP_API_KEY`.
     """
+    queryset = ComputationalResult.public_objects.all()
+    serializer_class = ComputationalResultSerializer
```

Comment:
If I'm not mistaken, you don't need to set `serializer_class` if you're using `get_serializer_class()`

```
Typically, you must either set this attribute, or override the get_serializer_class() method.
```